kDrive/AppRouter.swift

```diff
@@ -0,0 +1,428 @@
+/*
+ Infomaniak kDrive - iOS App
+ Copyright (C) 2024 Infomaniak Network SA
+
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import InfomaniakCore
+import InfomaniakCoreUI
+import InfomaniakDI
+import kDriveCore
+import kDriveResources
+import SafariServices
+import UIKit
+import VersionChecker
+
+/// Something that can navigate to specific places of the kDrive app
+public protocol RouterAppNavigable {
+    @MainActor func showMainViewController(driveFileManager: DriveFileManager)
```

Comment:
Annotating `@MainActor` for the UI navigation functions, and `async` for the actions that may present something, did remove a hang in my testing. Also prevents from miss using the UI bound methods.